Add `PolyInitialized` qualifier to the Initialization Checker (#494)
eisop/checker-framework: docs/CHANGELOG.md

Author: rewording is not done yet

**User-visible changes:**

The Initialization Checker supports new qualifier:

> * `@PolyInitialized`. By using the @PolyInitialized qualifier,

Reviewer: Why is this sentence split over three bullet points? It reads rather oddly.
The "acknowledges" is also odd wording. At compile time we don't have objects, so that's also not the word you want to use here. Please try rewording this paragraph.

Author: OK. But the manual still uses 'object'(e.g. The Initialization Checker determines whether an object is initialized or not. For any object that is not fully initialized, the Nullness Checker treats its fields as possibly-null — even fields annotated as @NonNull), should we change that as well? @wmdietl

Author: The format is changed. I have tried to change the words here. 

I am wondering what are the terminology at compile time and at run time for most java programs. Is there any reference for that?

> The Initialization Checker supports new qualifier: `@PolyInitialized`.
> 
> By using the `@PolyInitialized` qualifier, the Initialization Checker indicates that the field or class has a set of
> potential values, and the code using it must handle each possibility correctly.

Reviewer: ```suggestion
The Initialization Checker supports the new qualifier `@PolyInitialized` to express qualifier polymorphism.
```

**User-visible changes:**

> The Initialization Checker supports new qualifier:

Reviewer: Please make this a complete sentence that motivates why this is useful. Also, you should add a paragraph about the new qualifier to the manual.

Author: Already added.

**Closed issues:**

> eisop#376, eisop#494

Reviewer: ```suggestion
eisop#376, eisop#494.
```

**Closed issues:**

> eisop#376, eisop#494, eisop#532.

Reviewer: #494 is this PR, not an issue. I think we don't need this entry, there already is an entry above.

Author: Sure, thanks
